sonar-plugin-api/src/main/java/org/sonar/api/batch/sensor/issue/internal/DefaultIssue.java

```diff
@@ -156,9 +113,20 @@ public String key() {
   }
 
   @Override
+  public List<IssueLocation> locations() {
+    return ImmutableList.copyOf(this.locations);
+  }
+
+  @Override
+  public List<List<IssueLocation>> executionFlows() {
+    return ImmutableList.copyOf(this.executionFlows);
+  }
+
+  @Override
   public void doSave() {
     Preconditions.checkNotNull(this.ruleKey, "ruleKey is mandatory on issue");
     Preconditions.checkState(!Strings.isNullOrEmpty(key), "Fail to generate issue key");
+    Preconditions.checkState(locations.size() >= 1, "At least one location is mandatory on every issue");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use isEmpty() to check whether the collection is empty or not. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1155)